java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/memory/azurecognitivesearch/AzureCognitiveSearchRecord.java

```diff
@@ -0,0 +1,211 @@
+    private final String externalSourceName;
+
+    @SimpleField(isFilterable = false)
+    private final boolean isReference;
+
+    public AzureCognitiveSearchRecord(
+            @Nonnull String id,
+            @Nullable String text,
+            @Nullable String description,
+            @Nullable String additionalMetadata,
+            @Nullable String externalSourceName,
+            boolean isReference) {
+        this.id = id;
+        this.text = text;
+        this.description = description;
+        this.additionalMetadata = additionalMetadata;
+        this.externalSourceName = externalSourceName;
+        this.isReference = isReference;
+    }
+
+    /**
+     * Record ID. The record is not filterable to save quota, also SK uses only semantic search.
+     *
+     * @return Record ID.
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Content is stored here.
+     *
+     * @return Content is stored here, {@code null} if not set.
+     */
+    public String getText() {
+        return text;
+    }
+
+    /**
+     * Optional description of the content, e.g. a title. This can be useful when indexing external
+     * data without pulling in the entire content.
+     *
+     * @return Optional description of the content, {@code null} if not set.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Additional metadata. Currently, this is a String where you could store serialized data as
+     * JSON. In future the design might change to allow storing named values and leverage filters.
+     *
+     * @return Additional metadata, {@code null} if not set.
+     */
+    public String getAdditionalMetadata() {
+        return additionalMetadata;
+    }
+
+    /**
+     * Name of the external source, in cases where the content and the Id are referenced to external
+     * information.
+     *
+     * @return Name of the external source, in cases where the content and the Id are referenced to
+     *     external information, {@code null} if not set.
+     */
+    public String getExternalSourceName() {
+        return externalSourceName;
+    }
+
+    /**
+     * Whether the record references external information.
+     *
+     * @return {@code true} if the record references external information, {@code false} otherwise.
+     */
+    public boolean isReference() {
+        return isReference;
+    }
+
+    public static class Serializer extends StdSerializer<AzureCognitiveSearchRecord> {
+        public Serializer() {
+            this(null);
+        }
+
+        public Serializer(Class<AzureCognitiveSearchRecord> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(
+                AzureCognitiveSearchRecord azureCognitiveSearchRecord,
+                JsonGenerator jsonGenerator,
+                SerializerProvider serializerProvider)
+                throws IOException {
+            jsonGenerator.writeStartObject();
```

Comment:
Is there a need to do custom serialization, can it not be handled by annotations?

Reply:
That's odd. I recall you helped me with this. I'm trying 
```
    public AzureCognitiveSearchRecord(
             @JsonProperty("Id") String id,
             @JsonProperty("Text") String text,
             @JsonProperty("Description") String description,
             @JsonProperty("AdditionalMetadata") String additionalMetadata,
             @JsonProperty("ExternalSourceName") String externalSourceName,
             @JsonProperty("Reference") boolean isReference) {
```
But I get the exception that lead me to try the serializer/deserializer to begin with. 
```
java.lang.NullPointerException: Cannot invoke "com.fasterxml.jackson.databind.introspect.TypeResolutionContext.resolveType(java.lang.reflect.Type)" because "this._typeContext" is null
	at com.fasterxml.jackson.databind.introspect.AnnotatedMethod.getType(AnnotatedMethod.java:76)
```
What am I doing wrong?

Reply:
Latest commit removes the custom serialization, but I still have the method com.microsoft.semantickernel.connectors.memory.azurecognitivesearch.AzureCognitiveSearchRecord#searchFields

I cannot figure out why FieldBuilder.build(AzureCognitiveSearchRecord.class, null) fails (with the NPE given above).

Reply:
I think this is an issue in the Azure SDK and I've filed and issue: https://github.com/Azure/azure-sdk-for-java/issues/35584
For now, the code com.microsoft.semantickernel.connectors.memory.azurecognitivesearch.AzureCognitiveSearchRecord#searchFields will have to stand. But I'll update the comment about the NPE from FieldBuilder with a reference to the Azure SDK issue.